src/components/structures/RoomView.js

```diff
@@ -414,6 +420,7 @@ module.exports = React.createClass({
             this.setState({
                 room: room
             });
+            this._initTimeline(this.props);
```

Comment:
this doesn't feel right at all. wouldn't this trigger a reload if someone changes the room name? There must be some other way you can tell that the join has happened.

Also: `_initTimeline` returns a promise. If you're going to call it, you need to `.done()` it.